manager/controllers/package_detail.go

```diff
@@ -0,0 +1,118 @@
+package controllers
+
+import (
+	"app/base/database"
+	"app/base/models"
+	"app/base/utils"
+	"github.com/gin-gonic/gin"
+	"github.com/jinzhu/gorm"
+	"net/http"
+)
+
+func singlePackageLatestQuery(pkgName string) *gorm.DB {
+	return database.Db.Table("package pkg").Select("*").Joins("JOIN package_name pn ON pkg.name_id = pn.id").
+		Joins("JOIN advisory_metadata adv ON pkg.advisory_id = adv.id").
+		Where("pn.name = ?", pkgName).
+		Order("adv.public_date DESC").Limit(1)
+}
+
+func singlePackageEvraQuery(pkgName string, evra string) *gorm.DB {
+	return database.Db.Table("package").
+		Joins("JOIN package_name ON package.name_id = package_name.id").
+		Where("package_name.name = ?", pkgName).
+		Where("package.evra = ?", evra)
+}
+
+func getAdvisoryByID(advID int) (models.AdvisoryMetadata, error) {
+	var advisory models.AdvisoryMetadata
+	err := database.Db.Where("id = ?", advID).First(&advisory).Error
+	return advisory, err
+}
+
+type PackageDetailResponse struct {
+	Data SinglePackage `json:"data"`
+	Name string        `json:"name"`
+}
+
+type SinglePackage struct {
+	Package  models.Package          `json:"package"`
+	Advisory models.AdvisoryMetadata `json:"advisory"`
+}
+
+func packageLatestHandler(c *gin.Context, packageName string) {
+	query := singlePackageLatestQuery(packageName)
+
+	var packageInfo models.Package
+	err := query.Find(&packageInfo).Error
+	if err != nil {
+		LogAndRespError(c, err, "database error")
+		return
+	}
+
+	var advisory models.AdvisoryMetadata
+	err = database.Db.Where("id = ?", packageInfo.AdvisoryID).First(&advisory).Error
+	if gorm.IsRecordNotFoundError(err) {
+		LogAndRespNotFound(c, err, "advisory not found")
+		return
+	}
+
+	c.JSON(200, PackageDetailResponse{
+		Data: SinglePackage{
+			Package:  packageInfo,
+			Advisory: advisory,
+		},
+		Name: packageName,
+	})
+}
+
+func packageEvraHandler(c *gin.Context, nevra *utils.Nevra) {
+	packageName := nevra.Name
+	evra := nevra.EVRAString()
+
+	query := singlePackageEvraQuery(packageName, evra)
+
+	var packageInfo models.Package
+	err := query.Find(&packageInfo).Error
+	if err != nil {
+		LogAndRespError(c, err, "database error")
+		return
+	}
+
+	advisory, err := getAdvisoryByID(packageInfo.AdvisoryID)
+	if gorm.IsRecordNotFoundError(err) {
+		LogAndRespNotFound(c, err, "advisory not found")
+		return
+	}
+
+	c.JSON(200, PackageDetailResponse{
+		Data: SinglePackage{
+			Package:  packageInfo,
+			Advisory: advisory,
+		},
+		Name: packageName,
+	})
+}
+
+// @Summary Show me metadata of selected package
+// @Description Show me metadata of selected package
+// @ID LatestPackage
+// @Security RhIdentity
+// @Accept   json
+// @Produce  json
+// @Param    package_name    path    string   true "Package name"
```

Comment:
I think it should be package_param here. And please provide more info how the param should look like, there are some possibilities of usage (package_name - latest, nevra - exact version).